### PR TITLE
Fix pad_sequence decomposition to support padding_side parameter

### PR DESCRIPTION
**Pull request by bobrenjc93**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.14.0) (oldest at bottom):
* __->__ #179359

The upstream `torch.nn.utils.rnn.pad_sequence` supports a `padding_side` parameter (`'right'` or `'left'`) that controls whether shorter sequences are padded on the right (default) or left. The Python decomposition did not accept this parameter, causing errors when `torch.compile` traced code using `padding_side='left'`.

```python
# Before (wrong) - missing padding_side parameter, always pads right
register_decomposition(aten.pad_sequence.default)
def pad_sequence(sequences, batch_first=False, padding_value=0.0):
    ...
    row = aten.constant_pad_nd(
        currseq, dim_paddings + (0, max_len - currseq.size(0)), padding_value
    )

# After (correct) - respects padding_side parameter
register_decomposition(aten.pad_sequence.default)
def pad_sequence(sequences, batch_first=False, padding_value=0.0, padding_side="right"):
    ...
    pad_amount = max_len - currseq.size(0)
    if padding_side == "right":
        row = aten.constant_pad_nd(
            currseq, dim_paddings + (0, pad_amount), padding_value
        )
    else:
        row = aten.constant_pad_nd(
            currseq, dim_paddings + (pad_amount, 0), padding_value
        )
```

This fix adds the `padding_side` parameter to the decomposition, implementing left-padding by prepending zeros instead of appending, matching the eager-mode implementation.

### CODE REVIEW

**Review by eellison — APPROVED** [review]

**Comment by ZainRizvi** [comment]
@pytorchbot revert -c ignoredsignal "test_prelu_decomp_dtype_mismatch_error is failing on trunk"

The `linux-jammy-aarch64-py3.10 / test` is currently broken, but your new tests fail there too. Please double check to ensure your tests pass before relanding these PRs.

**Comment by ZainRizvi** [comment]
@pytorchbot revert -c ignoredsignal -m "test_prelu_decomp_dtype_mismatch_error is failing on trunk"

The `linux-jammy-aarch64-py3.10 / test` is currently broken, but your new tests fail there too. Please double check to ensure your tests pass before relanding these PRs.

image
